File: rectangulo.cpp
#include <iostream>

class Rectangulo {
private:
    double longitud;
    double ancho;

public:
    Rectangulo(double l, double a) {
        longitud = l;
        ancho = a;
    }

    double calcularArea() {
        return longitud * ancho;
    }

    double calcularPerimetro() {
        return 2 * (longitud + ancho);
    }

    void cambiarLongitud(double nueva_longitud) {
        std::cout << "Antigua longitud: " << longitud << std::endl;
        longitud = nueva_longitud;
        std::cout << "Nueva longitud: " << longitud << std::endl;
    }

    void cambiarAncho(double nuevo_ancho) {
        std::cout << "Antiguo ancho: " << ancho << std::endl;
        ancho = nuevo_ancho;
        std::cout << "Nuevo ancho: " << ancho << std::endl;
    }
};

int main() {

    Rectangulo miRectangulo(5, 10);

    double area = miRectangulo.calcularArea();
    std::cout << "Area del rectangulo: " << area << std::endl;

    double perimetro = miRectangulo.calcularPerimetro();
    std::cout << "Perimetro del rectangulo: " << perimetro << std::endl;

    miRectangulo.cambiarLongitud(7);
    miRectangulo.cambiarAncho(8);

    return 0;
}

File: rectangulo.py
class Rectangulo:
    def __init__(self, longitud, ancho):
        self.longitud = longitud
        self.ancho = ancho
    
    def calcularArea(self):
        area = self.longitud*self.ancho
        return f"El área del rectangulo es: {area}"

    def calcularPerimetro(self):
        perimetro = 2*self.longitud + 2*self.ancho
        return f"El perímetro del rectangulo es: {perimetro}"

    def cambiarLongitud(self, nueva_longitud):
        print(f"Antigua longitud: {self.longitud}")
        self.longitud = nueva_longitud
        return f"Nueva longitud: {self.longitud} "

    def cambiarAncho(self, nuevo_ancho):
        print(f"Antiguo ancho: {self.ancho}")
        self.ancho = nuevo_ancho
        return f"Nuevo ancho: {self.ancho}"
    
rectangle = Rectangulo(15, 12)

print(rectangle.calcularArea())
print(rectangle.calcularPerimetro())
print(rectangle.cambiarAncho(10))
print(rectangle.cambiarLongitud(20))
       
